src/main/java/io/cdap/plugin/http/source/common/pagination/page/JsonPage.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.http.source.common.pagination.page;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.cdap.plugin.http.source.common.BaseHttpSourceConfig;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Returns elements from json one by one by given json path.
+ */
+class JsonPage implements BasePage {
+  private final String insideElementJsonPathPart;
+  private final Iterator<JsonElement> iterator;
+  private final JsonObject json;
+  private final Map<String, String> fieldsMapping;
+
+  JsonPage(BaseHttpSourceConfig config, String responseBody) {
+    json = JSONUtil.toJsonObject(responseBody);
+    JSONUtil.JsonQueryResponse queryResponse = JSONUtil.getJsonElementByPath(json, config.getResultPath());
+    insideElementJsonPathPart = queryResponse.getUnretrivedPath();
+    iterator = queryResponse.getAsJsonArray().iterator();
```

Comment:
We should support the case that the path points to either an object or an array.

I imagine there could be situations where the structure is pretty complex so they want to just get a big object and process it more specially in a transform later in the pipeline.

Reply:
Done